File: Review-09-Java-Shapes/Example-2/src/test/java/shapes/TestShapeFactory.java
package shapes;

import org.junit.FixMethodOrder;
import org.junit.runners.MethodSorters;
import org.junit.Test;
import org.junit.Before;

import static org.junit.Assert.*;

import static org.hamcrest.CoreMatchers.*;
import org.hamcrest.core.IsNull;
import static org.hamcrest.number.IsCloseTo.closeTo;

import java.util.List;
import java.util.Arrays;

/**
 *   **This is technically a set of Integration Tests**
 *
 * 1 - Does this piece of code perform the operations
 *     it was designed to perform?
 *
 * 2 - Does this piece of code do something it was not
 *     designed to perform?
 *
 * 1 Test per mutator
 */
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestShapeFactory {

    private List<String> knownNames;

    private int numberKnown;

    @Before
    public void setUp()
    {
        this.knownNames = Arrays.asList("Circle",
                                        "Square",
                                        "Triangle",
                                        "Equilateral Triangle",
                                        "Right Triangle");

        this.numberKnown = this.knownNames.size();
    }

    /**
     * Create a known valid Shape
     */
    @Test
    public void testCreateShapeSuccess()
    {
        assertThat(ShapeFactory.createShape("Circle"), is(notNullValue()));
    }

    /**
     * Try to create a known invalid Shape
     */
    @Test
    public void testcreateShapeFailure()
    {
        assertThat(ShapeFactory.createShape("Lol Nope"), is(nullValue()));
    }

    /**
     * Create a known valid Shape
     */
    @Test
    public void testIsKnownSuccess()
    {
        assertThat(ShapeFactory.isKnown("Circle"), is(true));
    }

    /**
     * Try to create a known invalid Shape
     */
    @Test
    public void testIsKnownFailure()
    {
        assertThat(ShapeFactory.isKnown("Lol Nope"), is(false));
    }

    @Test
    public void testListKnown()
    {
        String knownStr = ShapeFactory.listKnown();

        for (String name : this.knownNames) {
            assertThat(knownStr, containsString(name));
        }
    }

    @Test
    public void testNumberKnown()
    {
        assertThat(ShapeFactory.numberKnown(), equalTo(this.numberKnown));
    }
}

File: Review-06-Python-Shapes/Example-6/tests/shapes/test_shape_factory.py
import copy

import pytest
from hamcrest import *

import shapes.shape_factory as shape_factory
from shapes.shape import Shape

"""
**This is technically a set of Integration Tests**

1 - Does this piece of code perform the operations
    it was designed to perform?

2 - Does this piece of code do something it was not
    designed to perform?

1 Test per mutator
"""


@pytest.fixture
def known_shape_names_and_number():
    known_names = (
        "Circle",
        "Square",
        "Triangle",
        "Equilateral Triangle",
        "Right Triangle",
    )

    number_known = len(known_names)

    yield known_names, number_known


def test_create_shape_success():
    """
    Create a known valid Shape
    """

    assert_that(shape_factory.create("Circle"), is_(not_none()))


def test_create_shape_failure():
    """
    Try to create a known invalid Shape
    """

    assert_that(shape_factory.create("Lol Nope"), is_(None))


def test_is_known_success():
    """
    Create a known valid Shape
    """

    assert_that(shape_factory.is_known("Circle"))


def test_is_known_failure():
    """
    Try to create a known invalid Shape
    """

    assert_that(shape_factory.is_known("Lol Nope"), is_(False))


def test_list_known(known_shape_names_and_number):
    known_names, _ = known_shape_names_and_number

    known_str = shape_factory.list_known()

    for name in known_names:
        assert_that(known_str, contains_string(name))


def test_number_known(known_shape_names_and_number):
    _, number_known = known_shape_names_and_number
    assert_that(shape_factory.number_known(), equal_to(number_known))
